src/anomalib/models/image/hugging_face_wrapper/lightning_model.py

```diff
@@ -0,0 +1,254 @@
+        )
+        self.processor = AutoProcessor.from_pretrained(model_path)
+
+    def _setup(self) -> None:
+        dataloader = self.trainer.datamodule.train_dataloader()
+        pre_images = self.collect_reference_images(dataloader)
+        self.pre_images = pre_images
+
+    def training_step(self, batch: dict[str, str | torch.Tensor], *args, **kwargs) -> dict[str, str | torch.Tensor]:
+        """Train Step of LLM."""
+        del args, kwargs  # These variables are not used.
+        # no train on llm
+        return batch
+
+    @staticmethod
+    def configure_optimizers() -> None:
+        """WinCLIP doesn't require optimization, therefore returns no optimizers."""
+        return
+
+    def validation_step(self, batch: dict[str, str | torch.Tensor], *args, **kwargs) -> dict:
+        """Validation Step of WinCLIP."""
+        self._setup()
+        del args, kwargs  # These variables are not used.
+        bsize = len(batch["image_path"])
+        out_list: list[str] = []
+        pred_list = []
+        for x in range(bsize):
+            o = "NO - default"
+            if self.k_shot > 0:
+                o = self._api_call_few_shot(batch["image_path"][x])
+            else:
+                o = self._api_call_zero_shot(batch["image_path"][x])
+            p = 0.0 if o.startswith("N") else 1.0
+            out_list.append(o)
+            pred_list.append(p)
+
+        batch["str_output"] = out_list
+        batch["pred_scores"] = torch.tensor(pred_list).to(self.device)
+        return batch
+
+    @property
+    def trainer_arguments(self) -> dict[str, int | float]:
+        """Set model-specific trainer arguments."""
+        return {}
+
+    @property
+    def learning_type(self) -> LearningType:
+        """The learning type of the model.
+
+        Llm is a zero-/few-shot model, depending on the user configuration. Therefore, the learning type is
+        set to ``LearningType.FEW_SHOT`` when ``k_shot`` is greater than zero and ``LearningType.ZERO_SHOT`` otherwise.
+        """
+        return LearningType.FEW_SHOT if self.k_shot else LearningType.ZERO_SHOT
+
+    def collect_reference_images(self, dataloader: DataLoader) -> list[str]:
+        """Collect reference images for few-shot inference.
+
+        The reference images are collected by iterating the training dataset until the required number of images are
+        collected.
+
+        Returns:
+            ref_images (Tensor): A tensor containing the reference images.
+        """
+        ref_images: list[str] = []
+        for batch in dataloader:
+            images = batch["image_path"][: self.k_shot - len(ref_images)]
+            ref_images.extend(images)
+            if self.k_shot == len(ref_images):
+                break
+        return ref_images
+
+    def _load_image(self, image_file: str) -> Image.Image:
+        return Image.open(image_file).convert("RGB")
+
+    def _api_call_zero_shot(self, image_path: str) -> str:
```

Comment:
Would it be possible for a user to configure the prompt if needed? Or does it have to be hardcoded?